app/services/account_searcher.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class AccountSearcher
+
+  include SearchHelper
+
+  def initialize(original_scope, query)
+    @original_scope = original_scope
+    @query = query
+  end
+
+  def valid?
+    @query.to_s.length >= 3
+  end
+
+  def results
+    owner_where_clause = <<-end_of_where
+      (
+        LOWER(users.first_name) LIKE :term
+        OR LOWER(users.last_name) LIKE :term
+        OR LOWER(users.username) LIKE :term
+        OR LOWER(CONCAT(users.first_name, users.last_name)) LIKE :term
+      )
+      AND account_users.user_role = :acceptable_role
+      AND account_users.deleted_at IS NULL
+    end_of_where
+
+    term = generate_multipart_like_search_term(@query)
+
+    # retrieve accounts matched on user for this facility
+    results = @original_scope.joins(account_users: :user).where(
```

Comment:
Similar to my comment above, I'd consider extracting this query logic and the query logic below into their own methods and just using the `results` method to deliver the results.

So you wind up with something like:
```
def results
    results = accounts_matched_on_user_query(owner_where_clause) += accounts_matched_on_number_query

    results.uniq
end
```

Reply:
I like these ideas.For this particular PR I opted to just extract the class and not really change the underlying code too much.  I've started doing refactoring on this class as I've begun work on Dartmouth's story around expanding which fields we're searching. I'm planning on extracting methods so the `results` method is something simple and readable like `owner_matches.or(account_number_matches).or(description_matches)` (this may not be exactly what it ends up as, but it's in the vein).